docs/contributing/testing-guidelines/unit-testing.md

```diff
@@ -0,0 +1,165 @@
+# Unit testing
+
+Unit testing is the first phase of testing and is used to validate units of source code such as classes and functions.
+Typically, a unit of code is tested by validating its output for various inputs.
+Unit testing helps ensure that the code behaves as intended and prevents accidental change of behavior.
+
+Autoware uses the `ament_cmake` framework to build and run tests.
+The same framework is also used to analyze the test results.
+
+`ament_cmake` provides several convenience functions to make it easy to register tests in a CMake-based package and to ensure that JUnit-compatible result files are generated.
+It currently supports a few different testing frameworks like `pytest`, `gtest`, and `gmock`.
+
+In order to prevent tests running in parallel from interfering with each other when publishing and subscribing to ROS topics,
+it is recommended to use commands from [`ament_cmake_ros`](https://github.com/ros2/ament_cmake_ros/tree/master/ament_cmake_ros/cmake) to run tests in isolation.
+
+See below for an example of using `ament_add_ros_isolated_gtest` with `colcon test`.
+All other tests follow a similar pattern.
+
+## Create a unit test with gtest
+
+In `my_cool_pkg/test`, create the `gtest` code file `test_my_cool_pkg.cpp`:
+
+```cpp
+#include "gtest/gtest.h"
+#include "my_cool_pkg/my_cool_pkg.hpp"
+TEST(TestMyCoolPkg, TestHello) {
+  EXPECT_EQ(my_cool_pkg::print_hello(), 0);
+}
+```
+
+For more examples of `gtest` features, see the [gtest repo](https://github.com/google/googletest).
```

Comment:
Changed this to an info box, since the sentence is not directly related to the instructions. 

```suggestion
!!! info

    For more examples of `gtest` features, see the [gtest repo](https://github.com/google/googletest).
```